### ISSUE TITLE
train issue: nan loss

### ISSUE THREAD

**Issue by Coldwon** [NWPUZhoufei/LDP-Net#7]
train的时候损失一直是nan，有人遇到同样的情况吗？

**Comment by lzjjia** [comment]
加1

**Comment by Coldwon** [comment]
> 加1

你好，不知道你解决了没有，我的情况是发现self_image_loss和cross_image_loss这两个损失会变成inf，也就是说计算的时候log里面有0，在这两个损失计算之前，加上pred_query_set_aug = torch.clamp(pred_query_set_aug, min=1e-9, max=1e7)和pred_query_set_local = torch.clamp(pred_query_set_local, min=1e-9, max=1e7)限制最小范围，希望可以帮到你

**Comment by Coldwon** [comment]
> 加1

@lzjjia 你好，不知道你解决了没有，我的情况是发现self_image_loss和cross_image_loss这两个损失会变成inf，也就是说计算的时候log里面有0，在这两个损失计算之前，加上pred_query_set_aug = torch.clamp(pred_query_set_aug, min=1e-9, max=1e7)和pred_query_set_local = torch.clamp(pred_query_set_local, min=1e-9, max=1e7)限制最小范围，希望可以帮到你

**Comment by lzjjia** [comment]
@Coldwon 你好，我已经解决了，但是也感谢您的分享。我私信问了论文作者，作者说可能随机crops的时捕获到的是背景图，所以损失是nan

**Comment by Coldwon** [comment]
@lzjjia 好的。可以请教你是怎么解决的吗？

**Comment by healer996** [comment]
@lzjjia 你好，可以请问一下你是怎么解决的吗，感谢
